### PR TITLE
Default-off warning if function shadows union case

### PR DESCRIPTION
**Pull request by Martin521**
Alternative fix for #15559: opt-in warning

### CODE REVIEW

File: src/Compiler/FSComp.txt [hunk 1]
```diff
@@ -1724,3 +1724,4 @@ featureUnmanagedConstraintCsharpInterop,"Interop between C#'s and F#'s unmanaged
 3578,chkCopyUpdateSyntaxInAnonRecords,"This expression is an anonymous record, use {{|...|}} instead of {{...}}."
 3579,alwaysUseTypedStringInterpolation,"Interpolated string contains untyped identifiers. Adding typed format specifiers is recommended."
 3580,tcUnexpectedFunTypeInUnionCaseField,"Unexpected function type in union case field definition. If you intend the field to be a function, consider wrapping the function signature with parens, e.g. | Case of a -> b into | Case of (a -> b)."
+3582,tcInfoIfFunctionShadowsUnionCase,"This is a function definition that shadows a union case. If this is what you want, ignore or suppress this warning. If you want it to be a union case deconstruction, add parantheses."
```
Comment (smoothdeveloper) [inline src/Compiler/FSComp.txt:1727]: small typo here: parantheses -> parentheses

Comment (Martin521) [reply src/Compiler/FSComp.txt]: Thanks. Fixed.

Comment (Martin521) [reply src/Compiler/FSComp.txt]: Introduced a typo in the commit message instead :-)

**Review by abonie — APPROVED** [review]

**Review by edgarfgp — APPROVED** [review]

**Review by T-Gro — APPROVED** [review]

**Comment by edgarfgp** [comment]
Great work @Martin521 👍🏻

**Pull request merged**